x/concentrated-liquidity/incentives_test.go

```diff
@@ -358,6 +368,30 @@ func (s *KeeperTestSuite) TestCreateAndGetUptimeAccumulatorValues() {
 }
 
 func (s *KeeperTestSuite) TestCalcAccruedIncentivesForAccum() {
+	incentiveRecordOneWithDifferentStartTime := createIncentiveRecord(
+		incentiveRecordOne.IncentiveDenom,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime.Add(10),
+		incentiveRecordOne.MinUptime,
+	)
+
+	incentiveRecordOneWithDifferentMinUpTime := createIncentiveRecord(
+		incentiveRecordOne.IncentiveDenom,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime,
+		testUptimeTwo,
+	)
+
+	incentiveRecordOneWithDifferentDenom := createIncentiveRecord(
+		testDenomTwo,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime,
+		incentiveRecordOne.MinUptime,
+	)
+
 	type calcAccruedIncentivesTest struct {
 		poolId               uint64
 		accumUptime          time.Duration
```

Comment:
Note: maybe we should make this value []time.Duration because;
1. different records can have different uptimes and currently we're constrained to only having one uptime here
2. Also, because of this i couldn't charge incentives on `two incentive records with same denom, different min up-time`, which i should we should change

Reply:
Clarified offline – two incentive records with the same denom and different uptimes get put on separate uptime accumulators, so since `calcAccruedIncentivesForAccum` only checks records for one uptime/uptime accum keeping this as `time.Duration` makes sense